File: app/src/main/java/io/github/lazymockbean/data/DependencyFinder.java
package io.github.lazymockbean.data;

import java.util.HashSet;
import java.util.Set;
import java.util.Stack;

public class DependencyFinder {
    private final Set<Class<?>> dependencyChecker = new HashSet<>();
    private final Stack<Class<?>> dependencyStack = new Stack<>();

    public void push(Class<?> clazz) {
        dependencyStack.push(clazz);
        dependencyChecker.add(clazz);
    }

    public Boolean isEmpty() {
        return dependencyStack.isEmpty();
    }

    public Class<?> pop() {
        return dependencyStack.pop();
    }

    public Boolean hasAlradyFound(Class<?> clazz) {
        return dependencyChecker.contains(clazz);
    }
}

File: app/src/main/kotlin/io/github/taesk/lazymockbean/data/DependencyFinder.kt
package io.github.taesk.lazymockbean.data

import java.util.Stack

class DependencyFinder() {
    private val dependencyChecker: HashSet<Class<*>> = hashSetOf()
    private val dependencyStack: Stack<Class<*>> = Stack<Class<*>>()

    fun push(clazz: Class<*>) {
        dependencyStack.push(clazz)
        dependencyChecker.add(clazz)
    }

    fun isEmpty(): Boolean {
        return dependencyStack.isEmpty()
    }

    fun pop(): Class<*> {
        return dependencyStack.pop()
    }

    fun hasAlradyFound(clazz: Class<*>): Boolean {
        return clazz in dependencyChecker
    }
}
